substrate/frame/support/procedural/src/pallet/expand/call.rs

```diff
@@ -245,6 +245,33 @@ pub fn expand_call(def: &mut Def) -> proc_macro2::TokenStream {
 			}
 		});
 
+	let deprecation_status = if let Some(call) = def.call.as_ref() {
+		crate::deprecation::get_deprecation(&quote::quote! {#frame_support}, &call.attrs)
+			.expect("Correctly parse deprecation attributes")
+	} else {
+		quote::quote! { #frame_support::__private::metadata_ir::DeprecationStatus::NotDeprecated }
+	};
+
+	let default_deprecation_info =
+		quote::quote! { #frame_support::__private::metadata_ir::DeprecationStatus::NotDeprecated }
+			.to_string();
+
+	let indexes: Vec<proc_macro2::TokenStream> = methods
+		.iter()
+		.filter_map(|x| {
+			let key = x.call_index;
+			let deprecation_status =
+				crate::deprecation::get_deprecation(&quote::quote! {#frame_support}, &x.attrs)
+					.expect("Correctly parse deprecation attributes");
+			if deprecation_status.to_string() == default_deprecation_info {
```

Comment:
maybe not ideal to compare between string here, the TokenStream can be equivalent while the string is different.